pkg/controller.v1beta1/experiment/experiment_controller.go

```diff
@@ -354,18 +355,23 @@ func (r *ReconcileExperiment) createTrials(instance *experimentsv1beta1.Experime
 		return err
 	}
 	var trialNames []string
+	var errStrings []string
 	for _, trial := range trials {
 		if err = r.createTrialInstance(instance, &trial); err != nil {
 			logger.Error(err, "Create trial instance error", "trial", trial)
+			errStrings = append(errStrings, err.Error())
 			continue
 		}
 		trialNames = append(trialNames, trial.Name)
 	}
+	// Failed to reconcile all trials
+	if len(trialNames) == 0 && len(errStrings) != 0 {
+		return fmt.Errorf(strings.Join(errStrings, "\n"))
```

Comment:
Yes, eventually Experiment controller will try to create Trials until we reach the `maxFailedTrialCount`: https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/experiment/util/status_util.go#L195-L202.

@henrysecond1 I think it is a great enhancement to record this error to the events, but do we want to check errors and returned Trials ? We should have clear requirement **"When Trial is Failed"**.

For example, in case of [this error](https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/experiment/experiment_controller_util.go#L89-L91) we don't want to stop reconcile. For example, Trial is already exist because of unsynchronised policy of Kubernetes controllers.

I can see the following solution:
- rename `createTrialInstance` to `getTrialInstance`.
- move [Trial Creation](https://github.com/kubeflow/katib/blob/5bb3cad7d9e594e7d953f98d7064c5f0bffcc65a/pkg/controller.v1beta1/experiment/experiment_controller_util.go#L88-L92) out of this function.
- if `getTrialInstance` fails, return error for `createTrials` function.
- if `r.Create` fails continue the loop.

Any other suggestion @henrysecond1 @gaocegege @johnugeorge ?